smdebug/core/hook.py

```diff
@@ -552,6 +561,13 @@ def _increment_step(self):
             self.mode_steps[ModeKeys.GLOBAL] = self.step
         self._collections_to_save_for_step = None
 
+    def should_save_tensor(self, tensor_name: str, collection_name: str) -> bool:
+        # Called in the internal AWS codebase to determine
+        # if a particular tensor value should be saved
+        if self._is_collection_being_saved_for_step(collection_name):
+            return True
+        return self.is_tensor_in_custom_collection(tensor_name)
```

Comment:
if collection is not saved for step , you just check is_tensor_in_custom_collection, shouldnt you check should_save_tensor_for_step ?

Also, what is the step val at this point ? Say this call is made before step 1, then what is the value of step right now in smdebug? 
Can there be off by one error?

Reply:
The call cannot be made before the step is incremented because training only starts after the `on_train_begin callback` is executed.